### ISSUE TITLE
TML Server not starting

### ISSUE THREAD

**Issue by kamilskrzypek4** [tModLoader/tModLoader#1942]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

Latest public-1.4-alpha

### OS

Windows

### Platform

GOG

### Description

Seems like GOGs version of TML server is trying to use Steam Workshop to find mods

### Log File

[server.log](https://github.com/tModLoader/tModLoader/files/7493048/server.log)

### Steps to reproduce

launch start-tModLoaderServer.bat
or
create server in-game

### Actual Behavior

When I start server bat file and pick no to steam prompt I have an error:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Terraria.Social.Steam.WorkshopHelper.UGCBased.Downloader.GetListOfSubscribedItemsPaths() in tModLoader\Terraria\Social\Steam\WorkshopHelper.cs:line 60
   at Terraria.Social.Steam.WorkshopHelper.UGCBased.Downloader.Refresh(WorkshopIssueReporter issueReporter) in tModLoader\Terraria\Social\Steam\WorkshopHelper.cs:line 108
   at Terraria.ModLoader.Core.ModOrganizer.FindMods() in tModLoader\Terraria\ModLoader\Core\ModOrganizer.cs:line 38
   at Terraria.ModLoader.Core.ModOrganizer.LoadMods(CancellationToken token) in tModLoader\Terraria\ModLoader\Core\ModOrganizer.cs:line 115
   at Terraria.ModLoader.ModLoader.Load(CancellationToken token) in tModLoader\Terraria\ModLoader\ModLoader.cs:line 122
   at Terraria.ModLoader.ModLoader.Reload() in tModLoader\Terraria\ModLoader\ModLoader.cs:line 191
   at Terraria.Main.DedServ() in tModLoader\Terraria\Main.cs:line 4061
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 209
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 173
   at MonoLaunch.Main(String[] args) in tModLoader\Terraria\MonoLaunch.cs:line 52
```

When I am tring to start server in game same thing occurs (logs are nealy the same), in game all I see is creating server

### Reproduction frequency

Always, with or without steam in tray

### Additional Information

I don't own terraria / TML on steam

**Comment by Solxanich** [comment]
Hey there,

Apologies for delayed response - I'm unable to reproduce this myself, but duly note a similar bug existed on earlier versions of tmodloader and was since corrected.

Could you confirm if this bug is still present on the most recent version of tModLoader AND if the computer you're testing on has Steam installed?
I'm suspect that the latter is related.

I don't see anything wrong code wise; all steam commands are appropriately conditioned

**Comment by Solxanich** [comment]
Fixed in #2045
